core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/log.go

```diff
@@ -0,0 +1,56 @@
+package logprovider
+
+import (
+	"encoding/hex"
+
+	ocr2keepers "github.com/smartcontractkit/chainlink-common/pkg/types/automation"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+)
+
+// LogSorter sorts the logs based on block number, tx hash and log index.
+// returns true if b should come before a.
+func LogSorter(a, b logpoller.Log) bool {
+	return LogComparator(a, b) > 0
+}
+
+// LogComparator compares the logs based on block number, log index.
+// tx hash is also checked in case the log index is not unique within a block.
+//
+// Returns:
+//
+//	-1 if a <  b
+//	 0 if a == b
+//	+1 if a >  b
+func LogComparator(a, b logpoller.Log) int {
+	if a.BlockNumber != b.BlockNumber {
+		return int(a.BlockNumber - b.BlockNumber)
+	}
+	logIndexDiff := a.LogIndex - b.LogIndex
+	if logIndexDiff == 0 {
+		return a.TxHash.Big().Cmp(b.TxHash.Big())
+	}
+	return int(logIndexDiff)
```

Comment:
nit. this can be > 1 or < -1 depending on index.

It might work in practice but violates the comment above